src/main/java/appeng/me/cluster/implementations/CraftingCPUCluster.java

```diff
@@ -443,38 +443,86 @@ private IGrid getGrid() {
         return null;
     }
 
-    private boolean canCraft(final ICraftingPatternDetails details, final Collection<IAEItemStack> condensedInputs) {
-        for (IAEItemStack g : condensedInputs) {
+    private boolean canCraft(final ICraftingPatternDetails details) {
+
+        if (!details.isCraftable()) {
+            // Processing patterns are relatively easy
+            for (IAEItemStack input : details.getInputs()) {
+                final IAEItemStack ais = this.inventory.extractItems(input.copy(), Actionable.SIMULATE,
+                        this.machineSrc);
+                final ItemStack is = ais == null ? ItemStack.EMPTY : ais.createItemStack();
+
+                if (is.isEmpty() || is.getCount() < input.getStackSize()) {
```

Comment:
This is something we should test if it's needed. Calling `createItemStack()` is a bit expensive and the condition does not make use of any `ItemStack` relevant functions. So it could just be replaced by ` if (ais == null || ais.getStackSize() < input.getStackSize())`. Just needs to potentially take into account that `ItemStack` is limited to int, while `IAEStack` uses a long

Reply:
Changed